src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -527,6 +522,14 @@ Unit* Aura::GetCaster() const
     return ObjectAccessor::GetUnit(*GetOwner(), GetCasterGUID());
 }
 
+GameObject* Aura::GetGameObjectCaster() const
```

Comment:
Change this to return WorldObject and rename to GetWorldObjectCaster

Reply:
done :)